runtime/gc_base/StringTable.cpp

```diff
@@ -224,16 +224,7 @@ stringComparatorFn(struct J9AVLTree *tree, struct J9AVLTreeNode *leftNode, struc
 	stu8Ptr = *((UDATA*) (leftNode+1)); 
 
 	/* Get at the String information  */
-	right_s = *(j9object_t *)(rightNode+1);
-
-	if (!isMetronome) {
-		/* Check if string was copy-forwarded.  Only do this on non-metronome since metronome re-uses the FORWARDED bit */
-		MM_ScavengerForwardedHeader forwardedHeader(right_s, extensions);
-		J9Object* forwardedPtr = forwardedHeader.getForwardedObject();
-		if (NULL != forwardedPtr) {
-			right_s = forwardedPtr;
-		}
-	}
+	right_s = J9WEAKROOT_OBJECT_LOAD_VM(javaVM, (j9object_t *)(rightNode+1));
```

Comment:
I'm slightly concerned about the performance implications of this change as this will now call thru:
```c
 MM_ReadBarrierVerifier::preWeakRootSlotRead(J9VMThread *vmThread, j9object_t *srcAddress)
 {
 	Assert_MM_true(vmThread->javaVM->internalVMFunctions->currentVMThread(vmThread->javaVM) == vmThread);
```
and that assert isn't free.

What performance testing do you plan to do for this?